docker/all-in-one/entrypoint.sh

```diff
@@ -206,19 +209,42 @@ function start_supervisor {
   /usr/bin/supervisord -c $SUPERVISOR_CONF
 }
 
+DELEGATED_ARCHIVE_PATH=/data/delegated-init.tar.gz
+
+function fetch_and_execute_delegated_payload {
+  curl -s --time-cond $DELEGATED_ARCHIVE_PATH -o $DELEGATED_ARCHIVE_PATH "$DELEGATED_INIT_LOCATION"
+
+  if [ ! -f $DELEGATED_ARCHIVE_PATH ]; then
+    echo "No delegated payload found, bailing"
+    return
+  fi
+
+  # only extract a valid archive
+  if [[ $(tar -tzf $DELEGATED_ARCHIVE_PATH) ]]; then
+    # TODO: (tom) Only extract newer tar archives
```

Comment:
Could we save the filemtime of the last extracted tarball in a persistent file and compare if current tarball filemtime is different?

Reply:
Have solved the time issue by adding a touch to the entryscript at build time
We could only be changing salt files, but this file will always definitively be in the archive.